### ISSUE TITLE
Minting rune

### ISSUE THREAD

**Issue by gregorym** [secretkeylabs/sats-connect#100]
Hi,

Is it possible to construct a transaction to mint a Rune?

**Comment by victorkirov** [comment]
Hi

Yes, you can construct a rune minting psbt and send it to the wallet to sign. The wallet supports signing transactions with OP_RETURN outputs, but it won't finalise them, so you'll need to get the final PSBT, finalise it, and broadcast it.

**Comment by gregorym** [comment]
Do you have an example by any chance?

**Comment by gregorym** [comment]
There's an example in this PR: https://github.com/secretkeylabs/sats-connect/pull/91/files

I will close this.

**Comment by victorkirov** [comment]
@gregorym that's a bit different. We are releasing a sats-connect with a testnet beta version exposing our minting and etching API a bit later today. It will be live in Mainnet at the halving. You're welcome to use it 😄 We'll be publishing the updated docs today with those endpoints. You get a lot of added extras like repeat mint functionality.

If you want to build a Mint PSBT yourself, it's a bit more involved and requires you to build the Rune OP_RETURN script yourself, but it is possible if you want to go that route.

**Comment by victorkirov** [comment]
Correction: The new sats-connect version is already released (v2.2.0). The docs just need an update, which will happen soooooon.

**Comment by gregorym** [comment]
@victorkirov Got it.
What would be the difference between using your API vs building a mint PSBT?

**Comment by victorkirov** [comment]
If you build your own PSBT, you'd have to figure out the relevant mint OP_RETURN script and stick it in the PSBT. You'd also most likely only do 1 mint txn at a time.

With our mint offering, you can just call the mint rune op with the details of what you want to mint, along with how many repeats of the rune you want to mint (You can currently mint 20 times with a single op, to be increased in the near future). You can also specify how much you want to charge for the mint op from your dapp as an input and which address to receive those fees to, and we'll send them as part of the process.

Basically, it's a convenience op that we provide devs so they can focus on the UI and let us handle the technical bits.

**Comment by fabienghr** [comment]
we have some issue coming from the class wallet which does not recognize the provider

we tried some workaround but didn’t succeed

if you have a demo test or i can provide the code we did would be perfect

**Comment by victorkirov** [comment]
@fabienghr could you please open a new ticket with details about the issue you are facing?
